frontend/src/component/releases/ReleasePlanTemplate/ReleasePlanTemplateAddStrategyForm.tsx

```diff
@@ -152,10 +152,31 @@ export const ReleasePlanTemplateAddStrategyForm = ({
             ? String(currentStrategy.parameters.stickiness)
             : 'default';
 
+    const segmentsMap = allSegments?.reduce(
+        (acc, segment) => {
+            acc[segment.id] = segment;
+            return acc;
+        },
+        {} as Record<string, ISegment>,
+    );
+
     useEffect(() => {
-        setSegments([]);
+        if (segmentsMap) {
+            setSegments(
+                (currentStrategy?.segments || []).map((segment) => {
+                    return segmentsMap[segment];
+                }),
+            );
+        }
     }, []);
 
+    useEffect(() => {
+        setCurrentStrategy((prev) => ({
+            ...prev,
+            segments: segments.map((segment) => segment.id),
+        }));
+    }, [JSON.stringify(segments)]);
```

Comment:
```suggestion
    }, segments);
```
why not just this? I think this is the state for React... maybe @nunogois will know better

Reply:
Ah thanks, I think you're right. This is not a deep object, so shouldn't have to use JSON.stringify here. I'll fix